README.md

```diff
@@ -1,6 +1,199 @@
 # plasmactl-bump
-Update the version of components that were updated in last commit
+Update the version of components that were updated in the last commit.
 
-### Excluded files
+### Excluded Files:
 - README.md
 - README.svg
+
+---
+
+### Work in Progress (WIP)
+
+**bump:** Updates the version of components that were modified after the last bump.  
+**Current bump logic:**
+1. Open the git repository.
+2. Check if the latest commit is not a bump commit.
+3. Collect a list of changed files until the previous bump commit is found (matching by author??). If the `last` option is passed, only take changes from the last commit. Prepare a map of resource objects to update.
+4. Get the short hash of the last commit.
+5. Iterate through the resource map and update the version of each.
+
+---
+
+### `bump --sync` or Propagation
+
+When resources are decoupled and stored in packages, there’s a need to preserve versions and maintain the current state of resources.
+`bump` works closely with `plasmactl compose`. The purpose of `compose` is to build the platform using resources from different places (domain repo, packages outside the repo).
+The purpose of `bump --sync` is to propagate the versions of a build to the current state.
```

Comment:
The purpose of bump --sync is to propagate the versions of changed resources to dependant resources in current build (post composition) while preserving history of earlier propagated versions.